// @aws-sdk/client-transcribe-streaming/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { resolveEventStreamConfig, getEventStreamPlugin } = require("@aws-sdk/middleware-eventstream");
const { resolveWebSocketConfig, getWebSocketPlugin } = require("@aws-sdk/middleware-websocket");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { resolveEventStreamSerdeConfig } = require("@smithy/core/event-streams");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultTranscribeStreamingHttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { GetMedicalScribeStream$, StartCallAnalyticsStreamTranscription$, StartMedicalScribeStream$, StartMedicalStreamTranscription$, StartStreamTranscription$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
const { getTranscribeStreamingPlugin } = require("@aws-sdk/middleware-sdk-transcribe-streaming");
__exportStar(require("./models/errors"), exports);
const { TranscribeStreamingServiceException } = require("./models/TranscribeStreamingServiceException");
exports.TranscribeStreamingServiceException = TranscribeStreamingServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "transcribe",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class TranscribeStreamingClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = resolveEventStreamConfig(_config_8);
        const _config_10 = resolveWebSocketConfig(_config_9);
        const _config_11 = resolveRuntimeExtensions(_config_10, configuration?.extensions || []);
        this.config = _config_11;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultTranscribeStreamingHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class GetMedicalScribeStreamCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetMedicalScribeStream", {})
    .n("TranscribeStreamingClient", "GetMedicalScribeStreamCommand")
    .sc(GetMedicalScribeStream$)
    .build() {
}

class StartCallAnalyticsStreamTranscriptionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        getEventStreamPlugin(config),
        getWebSocketPlugin(config, {
            headerPrefix: 'x-amzn-transcribe-',
        }),
        getTranscribeStreamingPlugin(config),
    ];
})
    .s("Transcribe", "StartCallAnalyticsStreamTranscription", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("TranscribeStreamingClient", "StartCallAnalyticsStreamTranscriptionCommand")
    .sc(StartCallAnalyticsStreamTranscription$)
    .build() {
}

class StartMedicalScribeStreamCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        getEventStreamPlugin(config),
        getWebSocketPlugin(config, {
            headerPrefix: 'x-amzn-transcribe-',
        }),
        getTranscribeStreamingPlugin(config),
    ];
})
    .s("Transcribe", "StartMedicalScribeStream", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("TranscribeStreamingClient", "StartMedicalScribeStreamCommand")
    .sc(StartMedicalScribeStream$)
    .build() {
}

class StartMedicalStreamTranscriptionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        getEventStreamPlugin(config),
        getWebSocketPlugin(config, {
            headerPrefix: 'x-amzn-transcribe-',
        }),
        getTranscribeStreamingPlugin(config),
    ];
})
    .s("Transcribe", "StartMedicalStreamTranscription", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("TranscribeStreamingClient", "StartMedicalStreamTranscriptionCommand")
    .sc(StartMedicalStreamTranscription$)
    .build() {
}

class StartStreamTranscriptionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        getEventStreamPlugin(config),
        getWebSocketPlugin(config, {
            headerPrefix: 'x-amzn-transcribe-',
        }),
        getTranscribeStreamingPlugin(config),
    ];
})
    .s("Transcribe", "StartStreamTranscription", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("TranscribeStreamingClient", "StartStreamTranscriptionCommand")
    .sc(StartStreamTranscription$)
    .build() {
}

const commands = {
    GetMedicalScribeStreamCommand,
    StartCallAnalyticsStreamTranscriptionCommand,
    StartMedicalScribeStreamCommand,
    StartMedicalStreamTranscriptionCommand,
    StartStreamTranscriptionCommand,
};
class TranscribeStreaming extends TranscribeStreamingClient {
}
createAggregatedClient(commands, TranscribeStreaming);

const ItemType = {
    PRONUNCIATION: "pronunciation",
    PUNCTUATION: "punctuation",
};
const ParticipantRole = {
    AGENT: "AGENT",
    CUSTOMER: "CUSTOMER",
};
const ContentRedactionOutput = {
    REDACTED: "redacted",
    REDACTED_AND_UNREDACTED: "redacted_and_unredacted",
};
const CallAnalyticsLanguageCode = {
    DE_DE: "de-DE",
    EN_AU: "en-AU",
    EN_GB: "en-GB",
    EN_US: "en-US",
    ES_US: "es-US",
    FR_CA: "fr-CA",
    FR_FR: "fr-FR",
    IT_IT: "it-IT",
    PT_BR: "pt-BR",
};
const Sentiment = {
    MIXED: "MIXED",
    NEGATIVE: "NEGATIVE",
    NEUTRAL: "NEUTRAL",
    POSITIVE: "POSITIVE",
};
const ClinicalNoteGenerationStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
const MedicalScribeNoteTemplate = {
    BEHAVIORAL_SOAP: "BEHAVIORAL_SOAP",
    BIRP: "BIRP",
    DAP: "DAP",
    GIRPP: "GIRPP",
    HISTORY_AND_PHYSICAL: "HISTORY_AND_PHYSICAL",
    PHYSICAL_SOAP: "PHYSICAL_SOAP",
    SIRP: "SIRP",
};
const ContentIdentificationType = {
    PII: "PII",
};
const ContentRedactionType = {
    PII: "PII",
};
const MedicalScribeParticipantRole = {
    CLINICIAN: "CLINICIAN",
    PATIENT: "PATIENT",
};
const MedicalScribeLanguageCode = {
    EN_US: "en-US",
};
const MedicalScribeMediaEncoding = {
    FLAC: "flac",
    OGG_OPUS: "ogg-opus",
    PCM: "pcm",
};
const MedicalScribeStreamStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PAUSED: "PAUSED",
};
const MedicalScribeVocabularyFilterMethod = {
    MASK: "mask",
    REMOVE: "remove",
    TAG: "tag",
};
const LanguageCode = {
    AF_ZA: "af-ZA",
    AM_ET: "am-ET",
    AR_AE: "ar-AE",
    AR_SA: "ar-SA",
    BE_BY: "be-BY",
    BG_BG: "bg-BG",
    BN_IN: "bn-IN",
    BS_BA: "bs-BA",
    CA_ES: "ca-ES",
    CKB_IQ: "ckb-IQ",
    CKB_IR: "ckb-IR",
    CS_CZ: "cs-CZ",
    CY_WL: "cy-WL",
    DA_DK: "da-DK",
    DE_CH: "de-CH",
    DE_DE: "de-DE",
    EL_GR: "el-GR",
    EN_AB: "en-AB",
    EN_AU: "en-AU",
    EN_GB: "en-GB",
    EN_IE: "en-IE",
    EN_IN: "en-IN",
    EN_NZ: "en-NZ",
    EN_US: "en-US",
    EN_WL: "en-WL",
    EN_ZA: "en-ZA",
    ES_ES: "es-ES",
    ES_MX: "es-MX",
    ES_US: "es-US",
    ET_ET: "et-ET",
    EU_ES: "eu-ES",
    FA_AF: "fa-AF",
    FA_IR: "fa-IR",
    FI_FI: "fi-FI",
    FR_CA: "fr-CA",
    FR_FR: "fr-FR",
    GL_ES: "gl-ES",
    GU_IN: "gu-IN",
    HE_IL: "he-IL",
    HI_IN: "hi-IN",
    HR_HR: "hr-HR",
    HT_HT: "ht-HT",
    HU_HU: "hu-HU",
    HY_AM: "hy-AM",
    ID_ID: "id-ID",
    IS_IS: "is-IS",
    IT_IT: "it-IT",
    JA_JP: "ja-JP",
    JV_ID: "jv-ID",
    KAB_DZ: "kab-DZ",
    KA_GE: "ka-GE",
    KK_KZ: "kk-KZ",
    KM_KH: "km-KH",
    KN_IN: "kn-IN",
    KO_KR: "ko-KR",
    LG_IN: "lg-IN",
    LT_LT: "lt-LT",
    LV_LV: "lv-LV",
    MK_MK: "mk-MK",
    ML_IN: "ml-IN",
    MR_IN: "mr-IN",
    MS_MY: "ms-MY",
    MY_MM: "my-MM",
    NE_NP: "ne-NP",
    NL_NL: "nl-NL",
    NO_NO: "no-NO",
    OR_IN: "or-IN",
    PA_IN: "pa-IN",
    PL_PL: "pl-PL",
    PS_AF: "ps-AF",
    PT_BR: "pt-BR",
    PT_PT: "pt-PT",
    RO_RO: "ro-RO",
    RU_RU: "ru-RU",
    SI_LK: "si-LK",
    SK_SK: "sk-SK",
    SL_SI: "sl-SI",
    SO_SO: "so-SO",
    SQ_AL: "sq-AL",
    SR_RS: "sr-RS",
    SU_ID: "su-ID",
    SV_SE: "sv-SE",
    SW_BI: "sw-BI",
    SW_KE: "sw-KE",
    SW_RW: "sw-RW",
    SW_TZ: "sw-TZ",
    SW_UG: "sw-UG",
    TA_IN: "ta-IN",
    TE_IN: "te-IN",
    TH_TH: "th-TH",
    TL_PH: "tl-PH",
    TR_TR: "tr-TR",
    UK_UA: "uk-UA",
    UZ_UZ: "uz-UZ",
    VI_VN: "vi-VN",
    ZH_CN: "zh-CN",
    ZH_HK: "zh-HK",
    ZH_TW: "zh-TW",
    ZU_ZA: "zu-ZA",
};
const MediaEncoding = {
    FLAC: "flac",
    OGG_OPUS: "ogg-opus",
    PCM: "pcm",
};
const MedicalContentIdentificationType = {
    PHI: "PHI",
};
const Pronouns = {
    HE_HIM: "HE_HIM",
    SHE_HER: "SHE_HER",
    THEY_THEM: "THEY_THEM",
};
const MedicalScribeSessionControlEventType = {
    END_OF_SESSION: "END_OF_SESSION",
};
const MedicalScribeTranscriptItemType = {
    PRONUNCIATION: "pronunciation",
    PUNCTUATION: "punctuation",
};
const PartialResultsStability = {
    HIGH: "high",
    LOW: "low",
    MEDIUM: "medium",
};
const Specialty = {
    CARDIOLOGY: "CARDIOLOGY",
    NEUROLOGY: "NEUROLOGY",
    ONCOLOGY: "ONCOLOGY",
    PRIMARYCARE: "PRIMARYCARE",
    RADIOLOGY: "RADIOLOGY",
    UROLOGY: "UROLOGY",
};
const VocabularyFilterMethod = {
    MASK: "mask",
    REMOVE: "remove",
    TAG: "tag",
};
const Type = {
    CONVERSATION: "CONVERSATION",
    DICTATION: "DICTATION",
};

exports.CallAnalyticsLanguageCode = CallAnalyticsLanguageCode;
exports.ClinicalNoteGenerationStatus = ClinicalNoteGenerationStatus;
exports.ContentIdentificationType = ContentIdentificationType;
exports.ContentRedactionOutput = ContentRedactionOutput;
exports.ContentRedactionType = ContentRedactionType;
exports.GetMedicalScribeStreamCommand = GetMedicalScribeStreamCommand;
exports.ItemType = ItemType;
exports.LanguageCode = LanguageCode;
exports.MediaEncoding = MediaEncoding;
exports.MedicalContentIdentificationType = MedicalContentIdentificationType;
exports.MedicalScribeLanguageCode = MedicalScribeLanguageCode;
exports.MedicalScribeMediaEncoding = MedicalScribeMediaEncoding;
exports.MedicalScribeNoteTemplate = MedicalScribeNoteTemplate;
exports.MedicalScribeParticipantRole = MedicalScribeParticipantRole;
exports.MedicalScribeSessionControlEventType = MedicalScribeSessionControlEventType;
exports.MedicalScribeStreamStatus = MedicalScribeStreamStatus;
exports.MedicalScribeTranscriptItemType = MedicalScribeTranscriptItemType;
exports.MedicalScribeVocabularyFilterMethod = MedicalScribeVocabularyFilterMethod;
exports.PartialResultsStability = PartialResultsStability;
exports.ParticipantRole = ParticipantRole;
exports.Pronouns = Pronouns;
exports.Sentiment = Sentiment;
exports.Specialty = Specialty;
exports.StartCallAnalyticsStreamTranscriptionCommand = StartCallAnalyticsStreamTranscriptionCommand;
exports.StartMedicalScribeStreamCommand = StartMedicalScribeStreamCommand;
exports.StartMedicalStreamTranscriptionCommand = StartMedicalStreamTranscriptionCommand;
exports.StartStreamTranscriptionCommand = StartStreamTranscriptionCommand;
exports.TranscribeStreaming = TranscribeStreaming;
exports.TranscribeStreamingClient = TranscribeStreamingClient;
exports.Type = Type;
exports.VocabularyFilterMethod = VocabularyFilterMethod;
